staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -219,8 +311,110 @@ func (q *delayingType) AddAfter(item interface{}, duration time.Duration) {
 	select {
 	case <-q.stopCh:
 		// unblock if ShutDown() is called
-	case q.waitingForAddCh <- &waitFor{data: item, readyAt: q.clock.Now().Add(duration)}:
+	case q.waitingForAddCh <- &waitFor{
+		data:    item,
+		readyAt: q.clock.Now().Add(duration),
+		options: ExpandedDelayingOptions{
+			PermitActiveAndWaiting: true,
+		}}:
+		// set PermitActiveAndWaiting to true so that the "active" and "waiting" queues act in isolation from each other.
```

Comment:
we can add that `this is the original behaviour of waiting queue and we are keeping it for backwards compatibility`

also can we move the comment up above the `PermitActiveAndWaiting` ?

Reply:
It might be good to mention something like this in the `AddAfter` documentation in the `DelayingInterface`

```go
	// AddAfter adds an item to the workqueue after the indicated duration has passed
        // This corresponds to AddWithOptions(item, ExpandedDelayingOptions{PermitActiveAndWaiting: true, WhenWaiting: TakeShorter, duration: duration})
	AddAfter(item interface{}, duration time.Duration)
```

Reply:
Added requested comments.